ta/arch/arm/ta.ld.S

```diff
@@ -8,14 +8,20 @@ OUTPUT_ARCH(aarch64)
 #endif
 
 PHDRS {
-	rodata PT_LOAD;
+	/*
+	 * Exec and rodata headers are hard coded to RX and RO
+	 * respectively. This is needed because the binary is relocatable
+	 * and the linker would automatically make any header writeable
+	 * that need to be updated during relocation.
+	 */
+	exec PT_LOAD FLAGS (5);		/* RX */
+	rodata PT_LOAD FLAGS (4);	/* RO */
 	rwdata PT_LOAD;
-	rodata2 PT_LOAD;
 	dyn PT_DYNAMIC;
 }
 
 SECTIONS {
-	.ta_head : {*(.ta_head)} :rodata
+	.ta_head : {*(.ta_head)} :exec
```

Comment:
Does `.ta_head` needs to be `:exec` instead of `:rodata`? I would think `:exec` should be applied to `.text` below instead.